python/MaterialXTest/tests_to_html.py

```diff
@@ -62,13 +63,12 @@ def main(args=None):
             print ("Number of glsl files " + str(len(glslFiles)) + " does not match number of osl files " +  str(len(oslFiles)) + " in dir: " + subdir)
             print ("GLSL list: " + str(glslFiles))
             print ("OSL files: " + str(oslFiles))
-            continue
-        elif len(glslFiles) > 0 and len(oslFiles) > 0:
+        if len(glslFiles) > 0 or len(oslFiles) > 0:
             fh.write("<h2>" + subdir + ":</h2><br>\n")
             fh.write("<table>\n")
-            for glslFile, oslFile in zip(glslFiles, oslFiles):
-                fullGlslPath = os.path.join(subdir, glslFile)
-                fullOslPath = os.path.join(subdir, oslFile)
+            for glslFile, oslFile in itertools.zip_longest(glslFiles, oslFiles):
```

Comment:
This should probably be improved further to match based on file name.

Reply:
Yes it would be good to perform a folder scan like this:
```
def walk_results(inputRoot, nameStr, outputList, fullfilePaths):
    for subdir, _, files in os.walk(inputRoot):
        for curFile in files:
          if curFile.endswith(".png") and not "diff.png" in curFile:
            if nameStr in curFile:
                outputList.append(curFile)
                fullfilePaths[curFile] = os.path.join(subdir, curFile)
```
and add in explicit reference location and comparison location(s) e.g.
```
   parser.add_argument('-i', '--inputdir', dest='inputdir', action='store', help='Input directory', default=".")
    parser.add_argument('-r', '--refdir', dest='groundrefdir', action='store', help='Ground Truth directory', default=".")
```
but this can be looked at afterwards as we do want to add additional languages and this script is mostly "quick-and-dirty" way based on 2 hard-coded folders.